packages/mason_logger/lib/src/progress.dart

```diff
@@ -53,37 +54,49 @@ class Progress {
   /// End the progress and mark it as completed.
   void complete([String? update]) {
     _stopwatch.stop();
-    final time =
-        (_stopwatch.elapsed.inMilliseconds / 1000.0).toStringAsFixed(1);
     _stdout.write(
-      '''\b${'\b' * (_message.length + 4)}\u001b[2K${lightGreen.wrap('✓')} ${update ?? _message} ${darkGray.wrap('(${time}s)')}\n''',
+      '''$_clearLn${lightGreen.wrap('✓')} ${update ?? _message} $_time\n''',
     );
     _timer.cancel();
   }
 
   /// End the progress and mark it as failed.
   void fail([String? update]) {
     _timer.cancel();
-    final time =
-        (_stopwatch.elapsed.inMilliseconds / 1000.0).toStringAsFixed(1);
-    _stderr.write(
-      '''\b${'\b' * (_message.length + 4)}\u001b[2K${red.wrap('✗')} ${update ?? _message} ${darkGray.wrap('(${time}s)')}\n''',
-    );
+    _stderr.write('$_clearLn${red.wrap('✗')} ${update ?? _message} $_time\n');
     _stopwatch.stop();
   }
 
+  /// Update the progress message.
+  void update([String? update]) {
```

Comment:
```suggestion
  void update(String update) {
```

Shouldn't the update be non-nullable? I can't think of a use-case in which you'd call update without providing a message. Wdyt?

Reply:
I considered it but figured some may want to keep the same original progress message with the elapsed time. Your changes are fine with me 🙂